pkg/lint/lintersdb/validator.go

```diff
@@ -75,3 +78,35 @@ func (v Validator) validatePresets(cfg *config.Linters) error {
 
 	return nil
 }
+
+func (v Validator) alternativeNamesDeprecation(cfg *config.Linters) error {
+	if v.m.cfg.InternalTest || v.m.cfg.InternalCmdTest || os.Getenv(logutils.EnvTestRun) == "1" {
+		return nil
+	}
+
+	altNames := map[string][]string{}
+	for _, lc := range v.m.GetAllSupportedLinterConfigs() {
+		for _, alt := range lc.AlternativeNames {
+			altNames[alt] = append(altNames[alt], lc.Name())
+		}
+	}
+
+	var names []string
+	names = append(names, cfg.Enable...)
+	names = append(names, cfg.Disable...)
+
+	for _, name := range names {
+		lc, ok := altNames[name]
+		if !ok {
+			continue
+		}
+
+		if len(lc) > 1 {
+			v.m.log.Warnf("The linter name %q is deprecated. It has been split into: %s.", name, strings.Join(lc, ", "))
```

Comment:
I may have missed something but I would suggest using "The linter named %q …"

The linter name is not deprecated, the linter is.

Or maybe a simpler version would be "The linter %q…"

Of course, as I said I may have missed something.

Reply:
It's the name of the linter, not the linter named x.

The linter is not deprecated.

It's just an alias.

Reply:
I reworded the 2 logs.

Reply:
You are right :+1: 

This case is only for `megacheck`/`staticcheck` because the deprecation never happens.
So yes, it's a "hidden" deprecation then the message should be defined accordingly.

The current system of alternative names is a problem with this split instead of deprecation (#3746)

The second case is the real use case of alternative names, and it is only about the name, not the linter.

So good catch!